scene/main/viewport.cpp

```diff
@@ -1265,7 +1265,12 @@ void Viewport::_gui_show_tooltip() {
 	Rect2 r(gui.tooltip_pos + tooltip_offset, gui.tooltip_popup->get_contents_minimum_size());
 
 	Window *window = gui.tooltip_popup->get_parent_visible_window();
-	Rect2i vr = window->get_usable_parent_rect();
+	Rect2i vr;
+	if (gui.tooltip_popup->is_embedded()) {
+		vr = Rect2i(Point2i(), window->size);
+	} else {
+		vr = window->get_usable_parent_rect();
+	}
 
```

Comment:
There are cases, where `window` is not the embedder of the popup. In these cases, some tooltips get positioned unexpected.
- Root window (embed_subwindows = true)
  - Window (embed_subwindows = true)
    - Button with tooltip

The tooltip of the button is displayed in the Root window, but in your suggestion, Window is used for calculating the position.

I suggest to use `Rect2i(Point2i(), gui.tooltip_popup->_get_embedder()->size);` instead of `Rect2i(Point2i(), window->size);` in your change.

Reply:
I hadn't thought about that, but quickly found an example with the sun/environment preview dialog, where tooltips do indeed position themselves unexpectedly. Changing this as per your suggestion.